pandas/core/reshape/pivot.py

```diff
@@ -580,29 +582,39 @@ def crosstab(
     b      0  1  0
     c      0  0  0
     """
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
```

Comment:
arer there tests that hit these cases?

Reply:
There are. FYI, I just moved this "defensive programming" block to the top.